### ISSUE TITLE
[build_response] Empty origin responses trigger an AssertionError in build_response

### ISSUE THREAD

**Issue by aph3rson** [abhinavsingh/proxy.py#559]
**Check FAQs**
Please check [Frequently Asked Questions](https://github.com/abhinavsingh/proxy.py#frequently-asked-questions) before opening a bug report.

**Describe the bug**
I'm building a plugin that has to modify the headers/body in both the request and response, and am using the built-in `proxy.plugin.modify_chunk_response` plugin as a template for doing so.

When calling an endpoint which returns a zero-length response, proxy.py throws an AssertionError. This appears to be due to `build_response` used in the example, which includes `self.body` in the assertion check. An origin could return a zero-length body, which would cause this assertion to fail.

**To Reproduce**
1. Modify the behavior in modify_chunk_response.py to not update the body, i.e. comment this line out: https://github.com/abhinavsingh/proxy.py/blob/525794b09652201d4058f9f809be41402a2c40e9/proxy/plugin/modify_chunk_response.py#L46

2. Run proxy.py with standard HTTP proxy settings, loading the aforementioned plugin:
    ```
    2021-06-01 16:25:19,838 - pid:15164 [I] load_plugins:334 - Loaded plugin proxy.http.proxy.HttpProxyPlugin
    2021-06-01 16:25:19,838 - pid:15164 [I] load_plugins:334 - Loaded plugin proxy.plugin.ModifyChunkResponsePlugin
    2021-06-01 16:25:19,839 - pid:15164 [I] listen:115 - Listening on 127.0.0.1:8899
    2021-06-01 16:25:19,842 - pid:15164 [D] start_workers:133 - Started acceptor#0 process 15196
    2021-06-01 16:25:19,845 - pid:15164 [D] start_workers:133 - Started acceptor#1 process 15197
    2021-06-01 16:25:19,848 - pid:15164 [D] start_workers:133 - Started acceptor#2 process 15198
    2021-06-01 16:25:19,851 - pid:15164 [D] start_workers:133 - Started acceptor#3 process 15199
    2021-06-01 16:25:19,851 - pid:15164 [I] start_workers:136 - Started 4 workers
    ```

3. Call an HTTP endpoint that returns no content in the response (e.g. a `204 No Content` response):
    ```
    $ curl --proxy localhost:9988 "http://httpbin.org/status/204" -H "accept: text/plain" -vvv
    *   Trying 127.0.0.1...
    * TCP_NODELAY set
    * Connected to localhost (127.0.0.1) port 8899 (#0)
    > GET http://httpbin.org/status/204 HTTP/1.1
    > Host: httpbin.org
    > User-Agent: curl/7.58.0
    > Proxy-Connection: Keep-Alive
    > accept: text/plain
    > 
    * Empty reply from server
    * Connection #0 to host localhost left intact
    curl: (52) Empty reply from server
    ```

4. Observe proxy.py's log output.
    ```
    2021-06-01 16:45:54,169 - pid:23651 [D] initialize:104 - Handling connection <socket.socket fd=18, family=AddressFamily.AF_INET, type=2049, proto=0, laddr=('127.0.0.1', 8899), raddr=('127.0.0.1', 47158)>
    2021-06-01 16:45:54,178 - pid:23651 [D] handle_readables:251 - Client is ready for reads, reading
    2021-06-01 16:45:54,178 - pid:23651 [D] recv:65 - received 140 bytes from client
    2021-06-01 16:45:54,179 - pid:23651 [D] connect_upstream:417 - Connecting to upstream httpbin.org:80
    2021-06-01 16:45:54,259 - pid:23651 [D] connect_upstream:422 - Connected to upstream httpbin.org:80
    2021-06-01 16:45:54,259 - pid:23651 [D] write_to_descriptors:157 - Server is write ready, flushing buffer
    2021-06-01 16:45:54,259 - pid:23651 [D] flush:92 - flushed 118 bytes to server
    2021-06-01 16:45:54,328 - pid:23651 [D] read_from_descriptors:179 - Server is ready for reads, reading...
    2021-06-01 16:45:54,328 - pid:23651 [D] recv:65 - received 225 bytes from server
    2021-06-01 16:45:59,214 - pid:23651 [E] run:363 - Exception while handling connection <socket.socket fd=18, family=AddressFamily.AF_INET, type=2049, proto=0, laddr=('127.0.0.1', 8899), raddr=('127.0.0.1', 47158)>
    Traceback (most recent call last):
      File "[snip]/proxy/http/handler.py", line 353, in run
        teardown = self.run_once()
      File "[snip]/proxy/http/handler.py", line 338, in run_once
        teardown = self.handle_events(readables, writables)
      File "[snip]/proxy/http/handler.py", line 158, in handle_events
        teardown = plugin.read_from_descriptors(readables)
      File "[snip]/proxy/http/proxy/server.py", line 212, in read_from_descriptors
        raw = plugin.handle_upstream_chunk(raw)
      File "[snip]/proxy/plugin/modify_chunk_response.py", line 47, in handle_upstream_chunk
        self.client.queue(memoryview(self.response.build_response()))
      File "[snip]/proxy/http/parser.py", line 257, in build_response
        assert self.code and self.version and self.body and self.type == httpParserTypes.RESPONSE_PARSER
    AssertionError
    ```

**Expected behavior**
A zero-length response is returned to the client.

**Version information**
 - OS: [e.g. iOS] Ubuntu Linux v. 18.04.5
 - Browser [e.g. chrome, safari] cURL 7.58.0
 - Device: [e.g. iPhone6] ...laptop?
 - proxy.py Version [e.g. 1.1.1] 2.3.1

**Additional context**
This should be a relatively easy change to this line: https://github.com/abhinavsingh/proxy.py/blob/d977fc71e8b02bfd26283ce2ccf8bee77ade16fa/proxy/http/parser.py#L257

Don't validate that there's a body, as `build_http_response()` already has a check for a zero-length body.
Alternatively, if plugin devs should be using `build_http_response()` directly, perhaps the example plugin should be updated.

**Comment by abhinavsingh** [comment]
@aph3rson Apologies for being away for too long.  This looks possible, will address this in coming release.  Thank you for reporting this.

**Comment by abhinavsingh** [comment]
Thank you again for reporting this.  Fixed in #688 , also updated the plugin example as requested :)
